### ISSUE TITLE
SDC_ATTR_COMPRESSION: New Bool Type

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/libSplash#184]
Follow-up to #153 #198 to use the current format version change: use the new, `h5py` compatible, types for bools with `SDC_ATTR_COMPRESSION` instead of `H5T_NATIVE_HBOOL` that are used right now.

**Comment by ax3l** [comment]
current type from `src/H5public.h`:
```c
/*
 * Boolean type.  Successful return values are zero (false) or positive
 * (true). The typical true value is 1 but don't bet on it.  Boolean
 * functions cannot fail.  Functions that return `htri_t' however return zero
 * (false), positive (true), or negative (failure). The proper way to test
 * for truth from a htri_t function is:
 *
 *  if ((retval = H5Tcommitted(type))>0) {
 *      printf("data type is committed\n");
 *  } else if (!retval) {
 *      printf("data type is not committed\n");
 *  } else {
 *      printf("error determining whether data type is committed\n");
 *  }
 */
typedef unsigned int hbool_t;
typedef int htri_t;
```

Btw tested via h5py "The typical true value is 1 but don't bet on it." -> it's not. it's just not 0.

**Comment by ax3l** [comment]
even if the full representation changes, since this value is only for pure information purposes (data set reads are handled compressed/uncompressed by hdf5 internally) we can go with a `minor version update` when we change it.

**Comment by ax3l** [comment]
Closed with #199
